### PR TITLE
[AUTO plugin] MULTI_DEVICE_PRIORITIES doesn't return device list by priority order from high to low

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - Correct the device list by priority order from high to low.
 - Remove GNA, CUDA, HPU, HDDL, NVIDIA from device list supported by AUTO/MULTI.

### Tickets:
 - 99669

### CODE REVIEW

File: src/plugins/auto/plugin.cpp [hunk 1]
```diff
@@ -416,6 +416,12 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
         }
         // reset the strDevices to support devices
         strDevices = "";
+        auto devicesWithPriority = GetValidDevice(supportDevices, networkPrecision);
```
Comment (peterchen-intel) [inline src/plugins/auto/plugin.cpp:419]: Need to exclude the case that MULTI_DEVICE_PRIORITIES is set and not all devices with "-" prefix.

Comment (peterchen-intel) [reply src/plugins/auto/plugin.cpp:419]: Was considered inside GetValidDevice

**Review by peterchen-intel — APPROVED** [review]

File: src/plugins/auto/plugin.cpp (hunk 1 above)
Comment (songbell) [inline src/plugins/auto/plugin.cpp:419]: let's add some comment to code, indicating why we have to call GetValidDevice here.

**Review by wangleis — APPROVED** [review]
Reviewed in meeting

**Comment by wangleis** [comment]
please update test cases also.

File: src/plugins/auto/plugin.cpp (hunk 1 above)
Comment (yangwang201911) [reply src/plugins/auto/plugin.cpp:420]: Done.

**Comment by peterchen-intel** [comment]
removal of hard code device list is being tracked on ticket 90102

File: src/plugins/auto/utils/config.hpp [hunk 2]
```diff
@@ -289,17 +289,7 @@ struct PluginConfig {
...
-                                                     "TEMPLATE",
-                                                     "VPUX",
-                                                     "MULTI",
-                                                     "HETERO",
-                                                     "CUDA",
-                                                     "NVIDIA",
-                                                     "HPU_GOYA",
-                                                     "mock"};
+    const std::set<std::string> _availableDevices =
+        {"AUTO", "CPU", "GPU", "TEMPLATE", "MYRIAD", "VPUX", "MULTI", "HETERO", "mock"};
```
Comment (nkogteva) [inline src/plugins/auto/utils/config.hpp:293]: After this change some tests started fail for `NVIDIA` plugin. @wangleis please take a look.

**Pull request merged**